build.gradle

```diff
@@ -1219,6 +1336,15 @@ project(':iceberg-spark3-runtime') {
     // relocate Antlr runtime and related deps to shade Iceberg specific version
     relocate 'org.antlr.v4.runtime', 'org.apache.iceberg.shaded.org.antlr.v4.runtime'
 
+    dependencies {
+      exclude 'org/antlr/runtime/**/*'
+      exclude 'org/abego/treelayout/**/*'
+      exclude 'com/ibm/icu/**/*'
+      exclude 'javax/json/**/*'
+      exclude 'org/glassfish/**/*'
+      exclude 'org/stringtemplate/**/*'
+    }
```

Comment:
It is suspicious to me that this is now necessary. I think it is because we are shading `runtimeClasspath` now rather than `compile`. I think that change is mostly correct because `implementation` dependencies are no longer transitive unless you look at the runtime classpath. But some extra dependencies must be leaking in from the other changes.

I think the right way to exclude these is to add them to the `implementation` excludes above, not to exclude specific packages.

Reply:
oh you're totally right. I moved stuff to the exclude list of the `implementation` configuration and we're getting the same results